apps/agent/README.md

```diff
@@ -0,0 +1,254 @@
+# Agent
+
+Agent is an Aragon app that has the ability to perform arbitrary calls to external contracts. It can be thought of as the external interface of a DAO. See our [guide](https://hack.aragon.org/docs/guides-use-agent) for concrete examples on how to use the Agent app.
+
+## 1. Functionalities
+
+### 1.1. Arbitrary call execution
+
+The following function can be used to execute arbitrary calls:
+
+- `execute(address _target, uint256 _ethValue, bytes _data)` - Protected by the `EXECUTE_ROLE` permission.
+
+If the sender has the right ACL permissions, it will execute an EVM call to `_target`, sending the specified ETH amount (in Wei) and `_data` as the calldata.
+
+- If the call reverts, it will revert forwarding the error data as the error data of the main call frame.
+- If the call succeeds, it will emit the `Execute` event logging the arguments the function was called with.
+
+Since any data can be passed as argument, it is important to note that the function could be a possible attack vector. Consider using `safeExecute` for all calls that don't require the transfer of funds. See the [Security](#14-security) section for further details.
+
+### 1.2. Signature handling
+
+
+
+### 1.3. Vault
+
+The Agent app is a superset of the [Vault](../vault/contracts/Vault.sol) contract, which means that it can store or transfer the tokens and Ether of the DAO in which it is installed. Specifically, the following functions are available: 
+
+ - `deposit(address _token, uint256 _value)` 
+ - `transfer(address _token, address _to, uint256 _value)` - Protected by the `TRANSFER_ROLE` permission.
+ - `balance(address _token)`
+
+More importantly, this property means that Agent is capable of using the DAO's funds to interact seemlessly with any contracts that require the sender to possess Ether or ERC-20 tokens (e.g. Exchanging tokens on [Uniswap](https://uniswap.io), participating in a [PoolTogether](https://www.pooltogether.com) lottery).
+
+### 1.4. Security
+
+Since the Agent app can manage tokens and Ether, it is crucial to always keep security in mind. In order to restrict unauthorized transfers and other abusive behaviors, a variant of the `execute` function is available that will prevent any attempt to transfer tokens included in a "protected tokens" list:
```

Comment:
```suggestion
Since the Agent app can manage tokens and Ether, it is crucial to always keep security in mind. In order to restrict unauthorized transfers and other abusive behaviors, a variant of the `execute()` function is available, and protected by a separate role, that will prevent any attempt to transfer tokens included in a "protected tokens" list:
```